src/jit/compiler.h

```diff
@@ -8106,12 +8106,14 @@ public :
         // callbacks are needed creates GT_RETURN(TYP_BYREF, op1 = Addr of RetBuf) for
         // methods with hidden RetBufArg.
         //
-        // TODO-AMD64-Unix - As per this ABI, addr of RetBuf needs to be returned by
-        // methods with hidden RetBufArg.  Right now we are special casing GT_RETURN
-        // of TYP_VOID in codegenxarch.cpp to generate "mov rax, addr of RetBuf".
-        // Instead we should consider explicitly materializing GT_RETURN of TYP_BYREF
-        // return addr of RetBuf in IR.
-        return compIsProfilerHookNeeded() && (info.compRetBuffArg != BAD_VAR_NUM);
+        // As per the System V ABI, the address of RetBuf needs to be returned by
+        // methods with hidden RetBufArg in RAX. In such case GT_RETURN is of TYP_BYREF,
+        // returning the address of RetBuf.
+        return 
+#if !defined(FEATURE_UNIX_AMD64_STRUCT_PASSING)
+                (compIsProfilerHookNeeded()) &&
+#endif // !defined(FEATURE_UNIX_AMD64_STRUCT_PASSING)
```

Comment:
(Minor) For better readability i would suggest to avoid use //!defined(....) if the amount of code enclosed under #if is the order of 1-10 lines.  

e.g. 
#if !defined(...)
#endif